api/v1alpha/istio-manifests/1.4/authz/allow-data-producers-policy.yaml

```diff
@@ -9,34 +9,46 @@ spec:
   rules:
   - to:
     - operation:
-        methods: ["GET,POST,PUT,DELETE"]
-        paths: ["/v1alpha/projects/$PROJECT_ID/accounts*"]
-    - operation:
-        methods: ["GET,POST,PUT,DELETE"]
-        paths: ["/v1alpha/projects/$PROJECT_ID/policies*"]
-    - operation:
-        methods: ["GET,POST,PUT,DELETE"]
-        paths: ["/v1alpha/projects/$PROJECT_ID/datasets*"]
-    - operation:
-        methods: ["POST"]
-        paths: ["/v1alpha/projects/$PROJECT_ID/admin*"]
-    - operation:
-        methods: ["POST"]
-        paths: ["/v1alpha/projects/$PROJECT_ID/auth:isDataProducer"]
-    - operation:
-        methods: ["GET"]
-        paths: ["/v1alpha/projects/$PROJECT_ID/products*"]
-    - operation:
-        methods: ["GET,POST"]
-        paths: ["/v1alpha/projects/$PROJECT_ID/procurements*"]
-    - operation:
-        methods: ["GET,POST"]
-        paths: ["/v1alpha/projects/$PROJECT_ID/spots*"]
-  - when:
-    - key: request.auth.claims[iss]
-      values:
-      - accounts.google.com
-      - https://accounts.google.com
-      - https://securetoken.google.com/$PROJECT_ID
+        methods:
+        - GET
+        paths:
+        - /v1alpha/projects/$PROJECT_ID/datasets?*
+        - /v1alpha/projects/$PROJECT_ID/procurements
+        - /v1alpha/projects/$PROJECT_ID/procurements?*
+        - /v1alpha/projects/$PROJECT_ID/views
+    - operation:
+        methods:
+        - POST
+        paths:
+        - /v1alpha/projects/$PROJECT_ID/accounts:reset
+        - /v1alpha/projects/$PROJECT_ID/admin:initSchema
+        - /v1alpha/projects/$PROJECT_ID/admin:syncResources
+        - /v1alpha/projects/$PROJECT_ID/auth:isDataProducer
+        - /v1alpha/projects/$PROJECT_ID/datasets
+        - /v1alpha/projects/$PROJECT_ID/procurements/approve
+    - operation:
+        methods:
+        - GET
+        - POST
+        paths:
+        - /v1alpha/projects/$PROJECT_ID/accounts
+        - /v1alpha/projects/$PROJECT_ID/policies
+    - operation:
+        methods:
+        - GET
+        - PUT
+        - DELETE
+        paths:
+        - /v1alpha/projects/$PROJECT_ID/accounts/*
+        - /v1alpha/projects/$PROJECT_ID/policies/*
+    - operation:
+        methods:
+        - GET
+        - POST
+        - PUT
+        - DELETE
+        paths:
+        - /v1alpha/projects/$PROJECT_ID/datasets/*
+    when:
```

Comment:
Does it make more sense to group resources together vs HTTP methods? I.E. GET /datasets; GET, POST, DELETE /datasets/*, etc. I think trying to visually identify where a rule operation is applied will be harder to debug the via GET /datasets, /products; GET, POST, DELETE /datasets/*, /products/* etc. e.g. This first [operation](https://github.com/GoogleCloudPlatform/datashare-toolkit/pull/353/files#diff-6c7814bd304dd17022946e5ae309ac0b65a55bf5c3ca78ed673c5e20d1bc4ae5R15) `GET /datasets*` is also included in the last operation at the bottom of the file [here](https://github.com/GoogleCloudPlatform/datashare-toolkit/pull/353/files#diff-6c7814bd304dd17022946e5ae309ac0b65a55bf5c3ca78ed673c5e20d1bc4ae5R51) `GET /datasets/*` WDYT?

Reply:
I'm not sure what's the easiest, I started by grouping resources, and then I narrowed it down to provide only the bare minimum of what's required. The GETS that you're referring to have slightly different paths.